br/pkg/lightning/backend/external/byte_reader.go

```diff
@@ -313,15 +317,16 @@ func (r *byteReader) reload() error {
 func (r *byteReader) closeConcurrentReader() (reloadCnt, offsetInOldBuffer int) {
 	r.logger.Info("drop data in closeConcurrentReader",
 		zap.Int("reloadCnt", r.concurrentReader.reloadCnt),
-		zap.Int("dropBytes", len(r.curBuf)-r.curBufOffset),
+		zap.Int("dropBytes", len(r.curBuf[r.curBufIdx])-r.curBufOffset),
```

Comment:
seems dropBytes need to add the totally unread curBuf `r.concurrentReader.bufSizePerConc * (len(r.curBuf) - 1 - r.curBufIdx)`